pkg/icons/icons.go

```diff
@@ -77,28 +78,24 @@ func Exists(filePath string) bool {
 	return false // File might not exist
 }
 
-func detectMIMEType(body io.ReadCloser) string {
-	buffer := make([]byte, 512)
-	_, err := body.Read(buffer)
-	if err != nil {
-		return ""
+// getExtension gets the appropriate file extension for the file contained in the
+// response body.
+func getExtension(resp *http.Response) (string, error) {
+	value, ok := resp.Header["Content-Type"]
```

Comment:
Can't we duplicate the body instead of only using the header? I don't think we can trust that the `Content-Type` header will be present for every icon